src/app/Fake.Tools.Squirrel/Squirrel.fs

```diff
@@ -0,0 +1,143 @@
+    {
+        ReleaseDir = ""
+        WorkingDir = None
+        BootstrapperExe = None
+        LoadingGif = None
+        SetupIcon = None
+        NoMsi = false
+        ToolPath = Tools.findToolInSubPath toolname ( Directory.GetCurrentDirectory() </> "tools" </> "Squirrel")
+        TimeOut = TimeSpan.FromMinutes 10.
+        SignExecutable = None
+        SigningKeyFile = None
+        SigningSecret = None })
+
+let private createSigningArgs (parameters : ReleasifyParams) =
+    new StringBuilder()
+    |> StringBuilder.appendWithoutQuotes "--signWithParams=\""
+    |> StringBuilder.appendWithoutQuotes "/a"
+    |> StringBuilder.appendIfSome parameters.SigningKeyFile (sprintf "/f %s")
+    |> StringBuilder.appendIfSome parameters.SigningSecret  (sprintf "/p %s")
+    |> StringBuilder.appendWithoutQuotes "\""
+    |> StringBuilder.toText
+
+let internal buildSquirrelArgs parameters nugetPackage =
+    new StringBuilder()
+    |> StringBuilder.appendIfNotNullOrEmpty nugetPackage "--releasify="
+    |> StringBuilder.appendIfNotNullOrEmpty parameters.ReleaseDir "--releaseDir="
+    |> StringBuilder.appendIfSome parameters.LoadingGif (sprintf "\"--loadingGif=%s\"")
+    |> StringBuilder.appendIfSome parameters.SetupIcon (sprintf "\"--setupIcon=%s\"")
+    |> StringBuilder.appendIfTrue parameters.NoMsi "--no-msi"
+    |> StringBuilder.appendIfSome parameters.BootstrapperExe (sprintf "\"--bootstrapperExe=%s\"")
+    |> StringBuilder.appendIfSome parameters.SignExecutable (fun _ -> createSigningArgs parameters)
+    |> StringBuilder.toText
+
+module internal ResultHandling =
+    let (|OK|Failure|) = function
+        | 0 -> OK
+        | x -> Failure x
+
+    let buildErrorMessage = function
+        | OK -> None
+        | Failure errorCode ->
+            Some (sprintf "Squirrel reported an error (Error Code %d)" errorCode)
+
+    let failBuildIfSquirrelReportedError =
+        buildErrorMessage
+        >> Option.iter failwith
+
+/// Creates a Squirrel installer for given NuGet package
+/// Will fail if Squirrel terminates with non-zero exit code.
+///
+/// ## Parameters
+///
+///  - `setParams` - Function used to manipulate the default `SquirrelParams` value.
+///  - `nugetPackage` - The package to create an installer for
+///
+/// ## Sample usage
+///
+///     Target.create "CreatePackage" (fun _ ->
+///         Squirrel.pack (fun p -> { p with WorkingDir = Some "./tmp" }) "./my.nupkg"
+///     )
+/// 
+/// ## Defaults for setParams
+///
+/// - `ReleaseDir` - `""`
+/// - `WorkingDir` - `None`
+/// - `BootstrapperExe` - `None`
+/// - `LoadingGif` - `None`
+/// - `SetupIcon` - `None`
+/// - `NoMsi` - `false`
+/// - `ToolPath` - The `squirrel.exe` path if it exists in a subdirectory of the current directory.
+/// - `TimeOut` - 10 minutes
+/// - `SignExecutable` - `None`
+/// - `SigningKeyFile` - `None`
+/// - `SigningSecret` - `None`
+let releasify (setParams: ReleasifyParams -> ReleasifyParams) (nugetPackage: string): unit =
```

Comment:
Ok so `Squirrel.releasify` (fine by me if that is a thing, just trying to show how it will look on the call side).
What about switching the arguments `releasify nugetPackage setParams`?

Reply:
I started with `run` or `pack` but all their documentation (and command line flags) call it like that... yes it's weird

Reply:
As for the switching I would prefer to keep that order, especially as it was already the one before (easier to port) and it's already the order on quite a lot of other modules

* https://github.com/fsharp/FAKE/blob/694f616c97fa242162cfd36db905d7df3156018f/src/app/Fake.DotNet.NuGet/Restore.fs#L128
* https://github.com/fsharp/FAKE/blob/98074fb18b75dca1ad4471c2bb2966e0369c0339/src/app/Fake.DotNet.MSBuild/MSBuild.fs#L471
* https://github.com/fsharp/FAKE/blob/694f616c97fa242162cfd36db905d7df3156018f/src/app/Fake.Documentation.DocFx/DocFx.fs#L54
* https://github.com/fsharp/FAKE/blob/694f616c97fa242162cfd36db905d7df3156018f/src/app/Fake.DotNet.Testing.OpenCover/OpenCover.fs#L133

Reply:
I guess the idea generally is to pipe the last argument, but I never actually see it used that way. Especially the `exec` in DocFx is spurious.
And yes this might be the first but still we should try to get it right in general and we might even update the api guideline based on this discussion ;)

Reply:
It was piped on the code I'm porting :) But I tend to remove all usage of single-pipe expressions from our codebase so it would have been gone.

I'll do the inversion

Reply:
I think consistency is more important than a slightly a better order of aguments. But if you want to start to change the order, it should be in the guidlines. DocFx, for example, is like this because i looked at the DotNet-Module.